### PR TITLE
Ft bubble sort ruby

### PR DESCRIPTION
**Pull request by Gabkings**
# Bubble-Sort

## Project Description

This project involves creating methods to sort a group of items in an array by using bubble sort techniques. A method was created to sort integers and another one to sort strings when the block is given.

## How it was Implemented

A method was created to accept an array of items in which each element is compared to the one next to it and they are swapped if the one on the left is larger than the one on the right. This continues until the array is eventually sorted.

## Built With

* RUBY
* VSCode

## Authors

👤 **Gabriel Gitonga**

- Github: [@Gabkings](https://github.com/Gabkings/BubbleSort)
- Linkedin: [linkedin](https://www.linkedin.com/in/gabriel-gitonga-b5a611183/)

## Contributing

Contributions, issues, and feature requests are welcome!
Feel free to check the [issues page](issues/).

### CODE REVIEW

File: bubble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    # subtract one because Ruby arrays are zero-index based
+    (array_length - 1).times do |i|
+      if array_to_sort[i] > array_to_sort[i + 1]
+        array_to_sort[i], array_to_sort[i + 1] = array_to_sort[i + 1], array_to_sort[i]
+        swapped = true
+      end
+    end
+    break unless swapped
+  end
+  puts array_to_sort
```
Comment (voscarmv) [inline bubble_sort.rb]: Please replace this line with a line returning the value of the sorted array.

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  while check
+    check = false
+    (0...sorted_array.length - 1).each do |i|
+      next unless yield(sorted_array[i], sorted_array[i + 1]).positive?
+
+      sorted_array[i], sorted_array[i + 1] = sorted_array[i + 1], sorted_array[i]
+      check = true
+    end
+  end
+  puts sorted_array
```
Comment (voscarmv) [inline bubble_sort.rb]: You should return the sorted array, not just print it. Please return the sorted array here.

**Review by voscarmv — CHANGES_REQUESTED** [review]
# Impressive work! 🖥️ 👀 ✨ 
## However, some changes are still required... 🧰 🔨 🔧

Your code seems to work, but for it to pass the automated tests, it must return the sorted array, not merely print it. 

Please read my code comments for details. 

### Submit these changes and request a new code review! 😄 👍 

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️

File: bubble_sort.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# rubocop: disable all
```
Comment (kitpao) [inline bubble_sort.rb]: - [ ] **MANDATORY** Please, do not disable rubocop linters, they are the guidelines for good coding practices in Microverse. Please let them do their work and fix all offenses that appear. You have 4 offenses
image

**Review by kitpao — CHANGES_REQUESTED** [review]
# 👋 Hello @Gabkings 
Great job fixing the required changes 🌟 now all the tests are passing 💯
You only have 1 mandatory change related to the linters, please check below 🔽 

**_Note:_** All the **[OPTIONAL]** are **not required** in order to approve your project but are personal suggestions to improve your coding experience

- [ ] **OPTIONAL** remember that in ruby, it is not necessary to write the word "return" because ruby automatically returns the last line of code. So you could just leave it as `array_to_sort` / `sorted_array`

### ♻️ Status: Your project requires 1 change ♻️
➡️ Please request another review once you are ready 📝 
___
##### 👉 If you have any doubts you can contact me in slack as @kitpao
##### 🌟  If you have feedback about my reviews I would appreciate if can leave a recommendation in LinkedIn 💌 or let me know in Slack if you see improvements to be done 😛 

🔗 __[Github](https://github.com/kitpao) | [Twitter](https://twitter.com/Kitpao1) | [LinkedIn](https://www.linkedin.com/in/kitzia-paola-vidal/) | [Medium](https://medium.com/@kitpao91)__

File: bubble_sort.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  while check
+    check = false
+    (0...array.length - 1).each do |i|
+      next unless yield(array[i], array[i + 1]).positive?
+
+      array[i], array[i + 1] = array[i + 1], array[i]
+      check = true
+    end
+  end
+  sorted_array
```
Comment (kitpao) [inline bubble_sort.rb]: - [ ] I see you changed the variable from sorted_array to array to reduce line length, but you forgot to change this line too. (It is not recognizing the variable)

File: .rubocop.yml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    Exclude:
+    - "Guardfile"
+    - "Rakefile"
+    
+    DisplayCopNames: true
+    
+    Layout/LineLength:
+    Max: 120
+    Metrics/MethodLength:
+    Max: 40
```
Comment (kitpao) [inline .rubocop.yml:11]: - [ ] These lines are different in the original, remember you should not modify the original rubocop file, please use the original, you can find it here: https://github.com/microverseinc/linters-config/blob/master/ruby/.rubocop.yml
image

File: .rubocop.yml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    Style/EachForSimpleLoop:
+    Enabled: false
+    Style/AndOr:
+    Enabled: false
+    Style/DefWithParentheses:
+    Enabled: false
+    Style/FrozenStringLiteralComment:
+    EnforcedStyle: never
+    Style/FrozenStringLiteralComment:
+        Enabled: false
```
Comment (kitpao) [inline .rubocop.yml:34]: - [ ] These lines are different in the original, remember you should not modify the original rubocop file, please use the original, you can find it here: https://github.com/microverseinc/linters-config/blob/master/ruby/.rubocop.yml
image

**Review by kitpao — CHANGES_REQUESTED** [review]
# 👋 Hi again @Gabkings 

You are almost there, it seems your changes triggered an error in one method so it is not passing the tests. And remember to use the original Rubocop file without modifications.

Please check below for details ⬇️ 

### ♻️ Status: Your project requires some additional changes ♻️
➡️ Please request another review once you are ready 📝 
___
##### 👉 If you have any doubts you can contact me in slack as @kitpao
##### 🌟  If you have feedback about my reviews I would appreciate if can leave a recommendation in LinkedIn 💌 or let me know in Slack if you see improvements to be done 😛 

🔗 __[Github](https://github.com/kitpao) | [Twitter](https://twitter.com/Kitpao1) | [LinkedIn](https://www.linkedin.com/in/kitzia-paola-vidal/) | [Medium](https://medium.com/@kitpao91)__

**Review by tirthajyoti-ghosh — APPROVED** [review]
# Approved ✅

## You have done an incredible job making this project 👏🧨💪! Just perfect 💯! 

### You have met all the requirements of the project. Go ahead and merge it! Onto the next project 🚀 !

Happy coding 👨‍💻 

<hr>

**Tirthajyoti Ghosh**

[GitHub](https://github.com/tirthajyoti-ghosh) | [LinkedIn](https://www.linkedin.com/in/tirthajyoti-ghosh/) | [Twitter](https://twitter.com/terrific_ghosh)

**Pull request merged**